From bbae57fd5ddcd65c034ef21586fab29600367f75 Mon Sep 17 00:00:00 2001
From: Asher Baker <asherkin@limetech.io>
Date: Sat, 20 Jul 2019 15:52:37 +0100
Subject: [PATCH 4/5] Support compilation on VS 2015

---
 src/common/windows/pe_util.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/windows/pe_util.h b/src/common/windows/pe_util.h
index 6c6b364f..80aba5e5 100644
--- a/src/common/windows/pe_util.h
+++ b/src/common/windows/pe_util.h
@@ -59,7 +59,7 @@ wstring GenerateDebugIdentifier(DWORD age, DWORD signature);
 
 // Converts |machine| enum value to the corresponding string used by Breakpad.
 // The enum is IMAGE_FILE_MACHINE_*, contained in winnt.h.
-constexpr const wchar_t* FileHeaderMachineToCpuString(WORD machine) {
+static inline const wchar_t* FileHeaderMachineToCpuString(WORD machine) {
   switch (machine) {
     case IMAGE_FILE_MACHINE_I386: {
       return L"x86";
-- 
2.25.1

